### PR TITLE
Display command line usage to console without requiring X Windows

### LINKED ISSUE #18602: bitcoin-qt --help requires an X display even though it is not used
bitcoin-qt will fail if it does not have access to a functioning X display, even though when run with --help it doesn't need one.

(I'm having a go at fixing this today)

### PR DESCRIPTION
**Pull request by rebroad**
Fixes #18602

### CODE REVIEW

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -431,6 +433,32 @@ int GuiMain(int argc, char* argv[])
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 #endif
 
+#if defined(WIN32)
+#else
```
Comment (luke-jr) [inline src/qt/bitcoin.cpp:437]:
```suggestion
#ifndef WIN32
```

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -431,6 +433,32 @@ int GuiMain(int argc, char* argv[])
...
+    node->setupServerArgs();
+    SetupUIArgs();
+    std::string error;
+    if (!gArgs.ParseParameters(argc, argv, error))
```
Comment (luke-jr) [inline src/qt/bitcoin.cpp:441]: Instead of duplicating code, perhaps move it to a new function.

**Review by luke-jr — CHANGES_REQUESTED** [review]
Concept ACK: It seems console-printed `--help` and errors in command-line options is the norm for other GUI applications as well.

**Comment by practicalswift** [comment]
Concept ACK

File: src/qt/bitcoin.cpp (hunk 2 above)
Comment (laanwj) [reply src/qt/bitcoin.cpp:441]: Agree. I really dislike duplicating all this code. I'm also not convinced that moving `ParseParameters` to another place in the initialization sequence won't interfere with some other part of the initialization sequence. It's quite a fragile sequence, carefully balanced to work for all use cases.

**Comment by MarcoFalke** [comment]
Can't this be worked around by setting `QT_QPA_PLATFORM=minimal` or so?

**Comment by fanquake** [comment]
This has a couple Concept ACKs, however needs follow up and comments addressing. Suggest moving to the GUI repo: https://github.com/bitcoin-core/gui.

**Pull request closed without merging**